trieste/acquisition/function/multi_objective.py

```diff
@@ -379,3 +384,245 @@ def _update_expected_improvement_fn(
                 isinstance(self._expected_improvement_fn, expected_hv_improvement), []
             )
             self._expected_improvement_fn.update(_partition_bounds)  # type: ignore
+
+
+class HippoPenalizationAcquisitionFunction(GreedyAcquisitionFunctionBuilder[ProbabilisticModel]):
+    r"""
+    HIPPO: HIghly Parallelizable Pareto Optimization
+
+    Builder of the acquisition function for greedily collecting batches by HIPPO
+    penalization in multi-objective optimization. The resulting acquistion function
+    takes in a set of pending points and returns a base acquisition function
+    penalized around those points.
+
+    Penalization is applied to the acquisition function multiplicatively. However, to
+    improve numerical stability, we perform additive penalization in a log space.
+    """
+
+    def __init__(
+        self,
+        objective_tag: str = OBJECTIVE,
+        base_acquisition_function_builder: AcquisitionFunctionBuilder[ProbabilisticModel]
+        | None = None,
+    ):
+        """
+        Initializes the HIPPO penalization function.
+
+        :param objective_tag: The tag for the objective data and model.
+        :param base_acquisition_function_builder: Base acquisition function to be
+            penalized. Defaults to Expected Hypervolume Improvement, also supports
+            its constrained version.
+        """
+        self._objective_tag = objective_tag
+        if base_acquisition_function_builder is None:
+            self._base_builder: AcquisitionFunctionBuilder[
```

Comment:
simpler like this? (or with `ProbabilisticModelType` if generic)

```suggestion
            self._base_builder = ExpectedHypervolumeImprovement[ProbabilisticModel]().using(self._objective_tag)
```